File: Accounts/tictacthor/workflows/src/main/java/net/corda/samples/tictacthor/accountUtilities/myGame.java
package net.corda.samples.tictacthor.accountUtilities;

import co.paralleluniverse.fibers.Suspendable;
import com.r3.corda.lib.accounts.contracts.states.AccountInfo;
import com.r3.corda.lib.accounts.workflows.services.AccountService;
import com.r3.corda.lib.accounts.workflows.services.KeyManagementBackedAccountService;
import net.corda.samples.tictacthor.states.BoardState;
import net.corda.core.flows.FlowException;
import net.corda.core.flows.FlowLogic;
import net.corda.core.flows.StartableByRPC;
import net.corda.core.flows.StartableByService;
import net.corda.core.node.services.vault.QueryCriteria;

import java.util.*;

@StartableByRPC
@StartableByService
public class myGame extends FlowLogic<BoardState>{

    private String whoAmI;
    public myGame(String whoAmI) {
        this.whoAmI = whoAmI;
    }

    @Override
    @Suspendable
    public BoardState call() throws FlowException {
        AccountService accountService = getServiceHub().cordaService(KeyManagementBackedAccountService.class);
        AccountInfo myAccount = accountService.accountInfo(whoAmI).get(0).getState().getData();
        QueryCriteria.VaultQueryCriteria criteria = new QueryCriteria.VaultQueryCriteria().withExternalIds(Arrays.asList(myAccount.getIdentifier().getId()));
        BoardState b = getServiceHub().getVaultService().queryBy(BoardState.class,criteria).getStates().get(0).getState().getData();
        return b;
    }
}

File: kotlin-samples/Accounts/tictacthor/workflows/src/main/kotlin/net/corda/samples/tictacthor/accountsUtilities/myGame.kt
package net.corda.samples.tictacthor.accountsUtilities


import co.paralleluniverse.fibers.Suspendable
import com.r3.corda.lib.accounts.workflows.accountService
import net.corda.samples.tictacthor.states.BoardState
import net.corda.core.flows.*
import net.corda.core.node.services.vault.QueryCriteria
import net.corda.core.utilities.ProgressTracker

// *********
// * Flows *
// *********
@InitiatingFlow
@StartableByRPC
class myGame(private val whoAmI:String) : FlowLogic<BoardState>() {
    override val progressTracker = ProgressTracker()

    @Suspendable
    override fun call():BoardState {

        val myAccount = accountService.accountInfo(whoAmI).single().state.data
        val criteria = QueryCriteria.VaultQueryCriteria(
                externalIds = listOf(myAccount.identifier.id)
        )

        return serviceHub.vaultService.queryBy(
                contractStateType = BoardState::class.java,
                criteria = criteria
        ).states.single().state.data
    }
}
